src/binSum.cpp

```diff
@@ -0,0 +1,319 @@
+#include <math.h>
+#include <Rmath.h>
+#include <RcppArmadillo.h>
+
+using namespace Rcpp;
+//#-------------------------------------------------------------------------#
+//' Function that computes a mean value for each bin
+//'
+//' @param x NumericVector - vector of values of a bin
+//' @param n intiger - number of bins
+//' @keywords internal
+// [[Rcpp::export]]
+NumericVector binMean(NumericVector x,int n) {
+  
+  int sz = x.size() ;// get the length of the input vector
+  NumericVector res(n);// create the output vector
+  double w_size=double(sz)/double(n); // window size can be a double
+  
+  // if the bins equals the vector size ,set the window size to 1
+  if(sz == n){
+    w_size=1;
+  }
+  
+  // if the bins number larger than vector size return zeros 
+  if(sz < n){
+    return res;
+  }
+  
+  double prev=0; // index for start positions over vector
+  int prev2 ;// integers for indices
+  int end2  ;
+  double end;
+  for(int i = 0; i< n; i++) {
+    end = prev + (w_size); //get the end index of the interval
+    prev2 = ceil(prev); // get the integer index for slices over vector
+    end2 = ceil(end);
+    prev = prev + w_size; // update the begining index of the slice
+    res[i] = std::accumulate(&x[prev2], &x[end2], 0.0)/(&x[end2]-&x[prev2]); //calculate the mean value of the bin
+  }
+  
+  return res;
+}
+
+// [[Rcpp::depends(RcppArmadillo)]]
```

Comment:
why does this depend on RcppArmadillo ?which part of the code depends on it ?

Reply:
Sorry, that was a part of the code from initial steps of implementing binMedian I've forgotten to remove